backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/CloneVmCommand.java

```diff
@@ -385,21 +400,23 @@ protected Guid getStoragePoolIdFromSourceImageContainer() {
      */
     @Override
     protected void lockEntities() {
-        TransactionSupport.executeInNewTransaction(() -> {
-            Snapshot snapshot = snapshotDao.get(getParameters().getSourceSnapshotId(), null, false);
-            getCompensationContext().snapshotEntityStatus(snapshot);
-            snapshotDao.updateStatus(snapshot.getId(), Snapshot.SnapshotStatus.LOCKED);
-            getCompensationContext().stateChanged();
-            return null;
-        });
-        freeLock();
+        // No need to lock snapshots for VMs with MBS disks
+        if (isNoManagedBlockDiskAttachedToVm()) {
```

Comment:
Early return would be better here.

Reply:
In both cases, the freeLock() should apply, not sure what return statement did you mean.

Reply:
In the code that I see `freeLock()` does not apply, if `isNoManagedBlockDiskAttachedToVm()` is `false`.

Reply:
Your eyes are better than mine on that time of the day :)
Ack